deploy/local_single_user_deploy.sh

```diff
@@ -237,123 +240,230 @@ stop_process() {
     pkill -f ${bin}
 }
 
-# Main functions for install, reset, and usage
-install() {
-    echo -e "${INFO}Installing ...${NC}"
+kill() {
+    echo -e "${DEBUG}Killing ...${NC}"
+    kill_process ${KEEPER_BIN}
+    kill_process ${GRAPHER_BIN}
+    kill_process ${VISOR_BIN}
+    echo -e "${DEBUG}Kill done${NC}"
+}
+
+check_build_directory() {
+    local deploy_dir=$(realpath "$(dirname "$0")")  # Get the absolute path of the script's directory
+    local build_dir="${deploy_dir}/../build"       # Navigate to x/build from x/deploy
+
+    build_dir=$(realpath "${build_dir}" 2>/dev/null || echo "")
+
+    echo -e "${DEBUG}Checking for the existence of the build directory: ${build_dir}${NC}"
+
+    if [ -d "${build_dir}" ]; then
+        echo -e "${DEBUG}Build directory found: ${build_dir}${NC}"
+    else
+        echo -e "${ERR}Build directory not found at: ${build_dir}${NC}"
+        echo "Please ensure the build process has been completed successfully before proceeding."
+        exit 1
+    fi
+}
+
+
+check_installation() {
     check_dependencies
     check_directories
     check_files
+}
+
+check_work_dir() {
+    # Check if WORK_DIR is set
+    if [[ -z "${WORK_DIR}" ]]; then
+        echo -e "${ERR}WORK_DIR is mandatory on this mode. Please provide it using the -w or --work-dir option.${NC}"
+        usage
+        exit 1
+    fi
+}
+
+# Main functions for install, reset, and usage__________________________________________________________________________
+build() {
+    echo -e "${INFO}Building ChronoLog...${NC}"
+
+    if [[ -n "$INSTALL_DIR" ]]; then
```

Comment:
if and else branch doing the same thing?

Reply:
True. Solved.